plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,336 @@
+package groundwork
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gwos/tcg/clients"
+	"github.com/gwos/tcg/milliseconds"
+	"github.com/gwos/tcg/transit"
+	"github.com/hashicorp/go-uuid"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+const (
+	defaultMonitoringRoute = "/api/monitoring?dynamic=true"
+)
+
+// Login and logout routes from Groundwork API
+const (
+	loginURL  = "/api/auth/login"
+	logoutURL = "/api/auth/logout"
+)
+
+var (
+	sampleConfig = `
+  ## HTTP endpoint for your groundwork instance.
+  endpoint = ""
+
+  ## Agent uuid for Groundwork API Server
+  agent_id = ""
+
+  ## Groundwork application type
+  app_type = ""
+
+  ## Username to access Groundwork API
+  username = ""
+
+  ## Password to use in pair with username
+  password = ""
+
+  ## Default display name for the host with services(metrics)
+  default_host = "default_telegraf"
+
+  ## Default service state [default - "SERVICE_OK"]
+  default_service_state = "SERVICE_OK"
+
+  ## The name of the tag that contains the hostname [default - "host"]
+  resource_tag = "host"
+`
+)
+
+type Groundwork struct {
+	Server              string `toml:"groundwork_endpoint"`
+	AgentID             string `toml:"agent_id"`
+	Username            string `toml:"username"`
+	Password            string `toml:"password"`
+	DefaultHost         string `toml:"default_host"`
+	DefaultServiceState string `toml:"default_service_state"`
+	ResourceTag         string `toml:"resource_tag"`
+	authToken           string
+}
+
+func (g *Groundwork) SampleConfig() string {
+	return sampleConfig
+}
+
+func (g *Groundwork) Init() error {
+	if g.Server == "" {
+		return errors.New("no 'groundwork_endpoint' provided")
+	}
+	if g.Username == "" {
+		return errors.New("no 'username' provided")
+	}
+	if g.Password == "" {
+		return errors.New("no 'password' provided")
+	}
+	if g.DefaultHost == "" {
+		g.DefaultHost = "telegraf"
+	}
+	if g.ResourceTag == "" {
+		g.ResourceTag = "host"
+	}
+	if g.DefaultServiceState == "" || !validStatus(g.DefaultServiceState) {
```

Comment:
I would error out if not a valid state is configured

Reply:
Done